pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2410,6 +2410,19 @@ func handleDNSImpl(ctxArg interface{}, key string,
 		log.Functionf("handleDNSImpl: ignoring %s", key)
 		return
 	}
+	if status.DPCKey == "" {
+		// Do not activate PhysicalIOAdapterList subscription until NIM receives
+		// a DPC and publishes corresponding DNS.
+		// NIM can publish DNS even before it receives first DPC. In such case
+		// DPCKey is empty.
+		// The goal is to avoid assigning network ports to PCIBack if they are going to be
+		// used for management purposes. This way we avoid doing unintended port assignment
+		// to PCIBack that would be shortly followed by a release, therefore mitigating
+		// the risk of race conditions between domainmgr and NIM.
+		log.Warnf("handleDNSImpl: DNS with empty DPCKey")
+		ctx.DNSinitialized = false
```

Comment:
Are there flows were we set DNSinitialized to true and then later end up setting it to false here?
I'm trying to understand any interaction with any wait for DNSinitialized in Run.

Reply:
domainmgr enable PhysicalIO subscription when `DNSinitialized` changes from `false` to `true`. The other way it has no effect. Actually, I could remove `ctx.DNSinitialized = false` and just return.